tests/suite/test_prometheus_endpoint.py

```diff
@@ -0,0 +1,148 @@
+import pytest
+import requests
+
+from kubernetes.client import V1ContainerPort
+
+from suite.resources_utils import (
+    ensure_connection_to_public_endpoint,
+    create_items_from_yaml,
+    create_example_app,
+    delete_common_app,
+    delete_items_from_yaml,
+    wait_until_all_pods_are_ready,
+    delete_secret,
+    create_secret_from_yaml,
+)
+from suite.yaml_utils import get_first_ingress_host_from_yaml
+from settings import TEST_DATA
+
+
+class IngressSetup:
+    """
+    Encapsulate the Smoke Example details.
+
+    Attributes:
+        ingress_host (str):
+    """
+
+    def __init__(self, ingress_host):
+        self.ingress_host = ingress_host
+
+
+@pytest.fixture(scope="class")
+def prometheus_setup(request, kube_apis, test_namespace):
+    print("------------------------- Deploy Prometheus Secret -----------------------------------")
+    prometheus_secret_name = create_secret_from_yaml(
+        kube_apis.v1, "nginx-ingress", f"{TEST_DATA}/prometheus/secret.yaml"
+    )
+
+    def fin():
+        delete_secret(kube_apis.v1, prometheus_secret_name, "nginx-ingress")
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture(scope="class")
+def enable_exporter_port(
+        cli_arguments, kube_apis, ingress_controller_prerequisites, ingress_controller
+) -> None:
+    """
+    Set containerPort for Prometheus Exporter.
+
+    :param cli_arguments: context
+    :param kube_apis: client apis
+    :param ingress_controller_prerequisites
+    :param ingress_controller: IC name
+    :return:
+    """
+    namespace = ingress_controller_prerequisites.namespace
+    port = V1ContainerPort(9113, None, None, "prometheus", "TCP")
+    print("------------------------- Enable 9113 port in IC -----------------------------------")
+    body = kube_apis.apps_v1_api.read_namespaced_deployment(ingress_controller, namespace)
+    body.spec.template.spec.containers[0].ports.append(port)
+
+    if cli_arguments["deployment-type"] == "deployment":
+        kube_apis.apps_v1_api.patch_namespaced_deployment(ingress_controller, namespace, body)
+    else:
+        kube_apis.apps_v1_api.patch_namespaced_daemon_set(ingress_controller, namespace, body)
+    wait_until_all_pods_are_ready(kube_apis.v1, namespace)
+
+
+@pytest.fixture(scope="class")
+def ingress_setup(request, kube_apis, ingress_controller_endpoint, test_namespace) -> IngressSetup:
```

Comment:
what is the rationale behind testing an Ingress resource in this test?

Reply:
merged with previous test.